File: Persona.java
import java.util.Objects;

public class Persona {
	private String nome;
	private String cognome;
	private String indirizzo;
	private String telefono;
	private Integer eta;
	
	public Persona(String n,String c,String i,String t,Integer e){
		this.nome=n;
		this.cognome=c;
		this.indirizzo=i;
		this.telefono=t;
		this.eta=e;
	}
	
	public void setNome(String n) {
		this.nome=n;
	}
	
	public void setCognome(String c) {
		this.cognome=c;
	}
	
	public void setIndirizzo(String i) {
		this.indirizzo=i;
	}
	
	public void setTelefono(String t) {
		this.telefono=t;
	}
	
	public void setEta(Integer e) {
		this.eta=e;
	}
	
	public String getNome() {
		return this.nome;
	}
	
	public String getCognome() {
		return this.cognome;
	}
	
	public String getIndirizzo() {
		return this.indirizzo;
	}
	
	public String getTelefono() {
		return this.telefono;
	}
	
	public Integer getEta() {
		return this.eta;
	}
	
	@Override
	public int hashCode() {
		return Objects.hash(this.telefono);
	}
	
	public String getStringHashCode() {
		return String.valueOf(this.hashCode());
	}
}

File: persona.cpp
#include "persona.h"

Persona::Persona(QString n,QString c,QString i,QString t,int e):nome(n),cognome(c),indirizzo(i),telefono(t),eta(e) {}

const QString Persona::getNome(){
    return this->nome;
}

const QString Persona::getCognome(){
    return this->cognome;
}

const QString Persona::getIndirizzo(){
    return this->indirizzo;
}

const QString Persona::getTelefono(){
    return this->telefono;
}

int Persona::getEta(){
    return this->eta;
}

void Persona::setNome(QString n){
    this->nome=n;
}

void Persona::setCognome(QString c){
    this->cognome=c;
}

void Persona::setIndirizzo(QString i){
    this->indirizzo=i;
}

void Persona::setTelefono(QString t){
    this->telefono=t;
}

void Persona::setEta(int e){
    this->eta=e;
}

bool Persona::operator==(const Persona& other) const{
    return this->telefono==other.telefono;
}
